localstack/utils/cloudformation/template_deployer.py

```diff
@@ -1125,6 +1174,14 @@ def delete_resource(resource_id, resources, stack_name):
             if 'NoSuchEntity' not in str(e):
                 raise
 
+    if res_type == 'AWS::EC2::VPC':
+        ec2_backend = ec2_backends[aws_stack.get_region()]
+        route_tables = ec2_backend.get_all_route_tables(filters={'vpc-id': res['PhysicalResourceId']})
```

Comment:
We should try to keep the logic in `template_deployer.py` as moto-independent as possible. Can we instead use something like this:
```
ec2_client = aws_stack.connect_to_service('ec2')
route_tables = ec2_client.describe_route_tables(
    Filters=[{'Name': 'vpc-id', 'Values': [res['PhysicalResourceId']]}])['RouteTables']
...
```